contracts/adapters/HegicAdapter.sol

```diff
@@ -352,19 +352,55 @@ contract HegicAdapter is IProtocolAdapter {
     }
 
     /**
-     * @notice Function to bulk claim rHEGIC2 rewards from liquidity utilization
+     * @notice Function to get rHEGIC2 rewards claimable from liquidity utilization
+     * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
+     * @param optionIDs is an array of all the option ids we want to claim rewards for
+     */
+    function rewardsClaimable(address rewardsAddress, uint256[] calldata optionIDs)
+        external
+        view
+        returns (uint256 rewardsAmount)
+    {
+        IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+
+        uint256 i = 0;
+
+        while (i < optionIDs.length && optionIDs[i] > 0) {
+          (, , , uint _amount, , uint _premium, , ) = rewardsContract.hegicOptions().options(optionIDs[i]);
+          if(!rewardsContract.rewardedOptions(optionIDs[i])){
+            rewardsAmount = rewardsAmount.add(_amount.div(100).add(_premium)
+            .mul(rewardsContract.rewardsRate())
+            .div(1e8));
+          }
+          i += 1;
+        }
+    }
+
+    /**
+     * @notice Function to get rHEGIC2 rewards claimable from liquidity utilization
      * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
      * @param optionIDs is an array of all the option ids we want to claim rewards for
      */
     function claimRewards(address rewardsAddress, uint256[] calldata optionIDs)
         external
+        returns (uint256 rewardsAmount)
     {
         IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+
         uint256 i = 0;
-        while (i < optionIDs.length) {
-          rewardsContract.getReward(optionIDs[i]);
+
+        uint256 balanceBefore = rewardsContract.hegic().balanceOf(address(this));
+
+        while (i < optionIDs.length && optionIDs[i] > 0) {
+          try rewardsContract.getReward(optionIDs[i]) {} catch {}
```

Comment:
Suggestion for another case to test for:

1) Buy first instrument
2) Collect reward for the first time
3) Buy second instrument
4) Collect reward for a second time

Testing that it shouldn't revert on step 4, and we're collecting the correct amount of rewards.